src/cargo-kani/src/call_single_file.rs

```diff
@@ -54,6 +54,11 @@ impl KaniSession {
             if !args.contains(&t) {
                 args.push(t);
             }
+        } else {
+            if self.args.function != "main" {
```

Comment:
Please add a comment that explains why we're compiling the crate as a library if the function is not main.